compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -480,6 +481,17 @@ extern "C" LLVMTargetMachineRef LLVMRustCreateTargetMachine(
     // it prevents control flow from "falling through" into whatever code
     // happens to be laid out next in memory.
     Options.TrapUnreachable = true;
+    // But don't emit traps after other traps or no-returns unnecessarily.
+    std::string wasm = "wasm";
+    if (std::string(TripleStr).substr(0, wasm.length()) != wasm) {
```

Comment:
```suggestion
    if (!Trip.isWasm()) {
```
Or something like that.